GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1567 @@
+﻿#region
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Threading.Tasks;
+using System.Data.Services.Client;
+using Microsoft.Azure.ActiveDirectory.GraphClient;
+using Microsoft.Azure.ActiveDirectory.GraphClient.Extensions;
+
+#endregion
+
+namespace GraphConsoleAppV3
+{
+    internal class Requests
+    {
+        private static string _graphAppObjectId;
+
+        public static async Task UserMode()
+        {
+            #region Setup Graph client for user
+
+            ActiveDirectoryClient client;
+
+            //*********************************************************************
+            // setup Microsoft Graph client for user...
+            //*********************************************************************
+            try
+            {
+                client = AuthenticationHelper.GetActiveDirectoryClientAsUser();
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("Acquiring a token failed with the following error: {0}", e.Message);
+                if (e.InnerException != null)
+                {
+                    //TODO: Implement retry and back-off logic per the guidance given here:http://msdn.microsoft.com/en-us/library/dn168916.aspx
+                    //InnerException Message will contain the HTTP error status codes mentioned in the link above
+                    Program.WriteError("Error detail: {0}", e.InnerException.Message);
+                }
+                Console.ReadKey();
+                return;
+            }
+
+            #endregion
+
+            Console.WriteLine("\nStarting user-mode requests...");
+            Console.WriteLine("\n=============================\n\n");
+
+
+            ITenantDetail tenantDetail = await GetTenantDetails(client);
+            User signedInUser = await GetSignedInUser(client);
+            await UpdateSignedInUsersPhoto(signedInUser);
+
+            Console.WriteLine("\nSearching for any user based on UPN, DisplayName, First or Last Name");
+            Console.WriteLine("\nPlease enter the user's name you are looking for:");
+            string searchString = Console.ReadLine();
+            await PeoplePickerExample(client, searchString);
+
+            User retrievedUser = await SearchForUserByUpn(client,
```

Comment:
>SearchForUserByUpn [](start = 39, length = 18)

This can return null User object. The call to ManipulateExtensionProperty and PrintExtensionProperty will be skipped if thats the case.